sql/analyzer/resolve_variables.go

```diff
@@ -231,8 +231,12 @@ func resolveSystemOrUserVariable(ctx *sql.Context, a *Analyzer, col column) (sql
 		a.Log("resolved column %s to session system variable", col)
 		return expression.NewSystemVar(varName, sql.SystemVariableScope_Session), true, nil
 	case sqlparser.SetScope_User:
+		t, _, err := ctx.GetUserVariable(ctx, varName)
```

Comment:
How does this behave when the user var isn't defined? I don't see any tests for this in `variable_queries.go`. Are there tests elsewhere you know of?

Reply:
Good question. I didn't see any tests for uninitialized user vars either, so I added a new ScriptTest to cover that. The behavior matches up to MySQL's documented behavior – referenced uninitialized user vars always return a null value.